src/main/java/com/minecolonies/coremod/blocks/BlockShingleSlab.java

```diff
@@ -0,0 +1,241 @@
+package com.minecolonies.coremod.blocks;
+
+import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.coremod.creativetab.ModCreativeTabs;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.properties.IProperty;
+import net.minecraft.block.properties.PropertyEnum;
+import net.minecraft.block.state.BlockStateContainer;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Locale;
+
+/**
+ * Decorative block
+ */
+public class BlockShingleSlab extends AbstractBlockMinecoloniesDirectional<BlockShingleSlab>
+{
+    /**
+     * The bounding box of the slab.
+     */
+    protected static final AxisAlignedBB AABB_BOTTOM_HALF = new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 0.5D, 1.0D);
+
+    /**
+     * The variants of the shingle slab.
+     */
+    public static final PropertyEnum<ShingleSlabType> VARIANT = PropertyEnum.create("variant", ShingleSlabType.class);
+
+    /**
+     * The hardness this block has.
+     */
+    private static final float BLOCK_HARDNESS = 3F;
+
+    /**
+     * This blocks name.
+     */
+    private static final String BLOCK_NAME = "blockshingleslab";
+
+    /**
+     * The resistance this block has.
+     */
+    private static final float RESISTANCE = 1F;
+
+    /**
+     * Amount of connections with other shingle slabs.
+     */
+    private static final int NO_CONNECTIONS    = 0;
+    private static final int THREE_CONNECTIONS = 1;
+    private static final int TWO_CONNECTIONS   = 2;
+    private static final int ONE_CONNECTION    = 3;
+
+    /**
+     * Constructor for the TimberFrame
+     */
+    public BlockShingleSlab()
+    {
+        super(Material.WOOD);
+        initBlock();
+    }
+
+    /**
+     * initialize the block
+     * sets the creative tab, as well as the resistance and the hardness.
+     */
+    private void initBlock()
+    {
+        setRegistryName(BLOCK_NAME);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(Locale.ENGLISH), BLOCK_NAME));
+        setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        setHardness(BLOCK_HARDNESS);
+        setResistance(RESISTANCE);
+        this.setLightOpacity(255);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 255 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)